cluster-provision/images/vm-image-builder/create-containerdisk.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+set -exuo pipefail
+
+SCRIPT_PATH=$(dirname "$(realpath "$0")")
+
+export CUSTOMIZE_IMAGE_SCRIPT=${CUSTOMIZE_IMAGE_SCRIPT:-"${SCRIPT_PATH}/customize-image.sh"}
+
+function customize_image() {
+  local source_image=$1
+  local os_variant=$2
+  local customized_image=$3
+  local cloud_config=$4
+
+  # Backup the VM image and pass copy of the original image
+  # in case customizing script fail.
+  vm_image_copy="$(dirname "${customized_image}")/copy-${source_image}"
+  cp "${source_image}" "${vm_image_copy}"
+
+  # TODO: convert this script and its dependencies to container
+  ${CUSTOMIZE_IMAGE_SCRIPT} "${vm_image_copy}" "${os_variant}" "${customized_image}" "${cloud_config}"
+
+  # Backup no longer needed.
+  rm -f "${vm_image_copy}"
+}
+
+function cleanup() {
+  if [ $? -ne 0 ]; then
+    rm -rf "${build_directory}"
+  fi
+
+  rm -f "copy-${VM_IMAGE}"
+}
+
+export IMAGE_NAME=${IMAGE_NAME:-example-fedora}
+export TAG=${TAG:-32}
+export OS_VARIANT=${OS_VARIANT:-fedora31}
+export CLOUD_CONFIG_PATH=${CLOUD_CONFIG_PATH:-"${SCRIPT_PATH}/example/cloud-config"}
+export VM_IMAGE_URL=${VM_IMAGE_URL:-$(cat "${SCRIPT_PATH}/example/image-url")}
+
+readonly VM_IMAGE="source-image.qcow2"
+readonly build_directory="${IMAGE_NAME}_build"
+readonly new_vm_image_name="provisioned-image.qcow2"
+
+trap 'cleanup' EXIT
```

Comment:
you said you need SIGINT right ?
also please consider adding SIGTERM, see
https://github.com/kubevirt/kubevirt/pull/4115#discussion_r484034330

Reply:
But we use `bash` here, not anything else.

Reply:
wouldnt it might run on prow one day ?
better be safe and robust i think, in any case

Reply:
It will probably run in a container and there is little chance we will not be fedora/centos base.

Reply:
imho until it has one click to run in a container i would suggest to be robust,
the change which is needed is minor and harmless from my point of view

Reply:
@oshoval , there is no robustness here, the script is limited to bash and it says so on the first line of the file.

Reply:
No need for `SIGINT`/`SIGTERM` for this PR scripts, I checked and it works as expected with `EXIT`
See this [disscution](https://github.com/kubevirt/kubevirtci/pull/428#discussion_r476229164)

Reply:
https://github.com/kubevirt/kubevirtci/pull/428#discussion_r476598843
you said the SIGINT is needed, if you double checked and its not needed, then remove but please update if possible so we know the behaviour.

SIGTERM is not needed because you have shebang of bash as Edy said.

Reply:
Alright I will add it , it was two weeks ago and all those threads are confusing :confused: